### PR TITLE
{nixos/,}clickshare-csc1: remove (qt4)

### PR DESCRIPTION
**Pull request by Yarny0**
###### Description of changes

Qt4 is on it's way out, according to https://github.com/NixOS/nixpkgs/pull/174634 .

There is already a commit dropping ClickShare in the pull request mentioned above, but we also need to drop the module, and -- as a courtesy to users -- should point to possible alternatives.

The pull request at hand removes ClickShare completely and appends the release notes with a link to a Barco (ClickShare producer) support page where they describe alternatives for newer clickshare base units.

###### Things done

- Built on platform(s)
  - [ ] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- [ ] For non-Linux: Is `sandbox = true` set in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://nixos.org/manual/nixpkgs/unstable/#sec-package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [ ] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [22.11 Release Notes (or backporting 22.05 Release notes)](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md#generating-2211-release-notes)
  - [x] (Package updates) Added a release notes entry if the change is major or breaking
  - [x] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
  - [x] (Release notes changes) Ran `nixos/doc/manual/md-to-db.sh` to update generated release notes
- [ ] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

### CODE REVIEW

File: nixos/doc/manual/release-notes/rl-2211.section.md [hunk 1]
```diff
@@ -47,6 +47,13 @@ In addition to numerous new and upgraded packages, this release has the followin
 
 - The `isPowerPC` predicate, found on `platform` attrsets (`hostPlatform`, `buildPlatform`, `targetPlatform`, etc) has been removed in order to reduce confusion.  The predicate was was defined such that it matches only the 32-bit big-endian members of the POWER/PowerPC family, despite having a name which would imply a broader set of systems.  If you were using this predicate, you can replace `foo.isPowerPC` with `(with foo; isPower && is32bit && isBigEndian)`.
 
+- The Barco ClickShare driver/client package `pkgs.clickshare-csc1` and its NixOS
+  configuration option `programs.clickshare-csc1.enable` have been removed,
+  as it requires `qt4`, which reached its end-of-life 2015 and
+  will no longer be supported by nipxkgs.
```
Comment (klemensn) [inline nixos/doc/manual/release-notes/rl-2211.section.md:53]: "nixpkgs"

File: pkgs/top-level/all-packages.nix [hunk 2]
```diff
@@ -25683,8 +25683,6 @@ with pkgs;
     inherit (xorg) libSM;
   };
 
-  clickshare-csc1 = callPackage ../applications/video/clickshare-csc1 { };
```
Comment (klemensn) [inline pkgs/top-level/all-packages.nix:25686]: You probably want to add a `clickshare-csc1 = throw "...";` line to `pkgs/top-level/aliases.nix` as well.

**Review by klemensn — CHANGES_REQUESTED** [review]

**Review by klemensn — APPROVED** [review]

**Comment by Yarny0** [comment]
Thanks, @klemensn , I pushed an updated commit.

File: nixos/doc/manual/release-notes/rl-2211.section.md [hunk 3]
```diff
@@ -47,6 +47,13 @@ In addition to numerous new and upgraded packages, this release has the followin
 
 - The `isPowerPC` predicate, found on `platform` attrsets (`hostPlatform`, `buildPlatform`, `targetPlatform`, etc) has been removed in order to reduce confusion.  The predicate was was defined such that it matches only the 32-bit big-endian members of the POWER/PowerPC family, despite having a name which would imply a broader set of systems.  If you were using this predicate, you can replace `foo.isPowerPC` with `(with foo; isPower && is32bit && isBigEndian)`.
 
+- The Barco ClickShare driver/client package `pkgs.clickshare-csc1` and its NixOS
+  configuration option `programs.clickshare-csc1.enable` have been removed,
+  as it requires `qt4`, which reached its end-of-life 2015 and
+  will no longer be supported by nixpkgs.
+  [According to Barco](https://www.barco.com/de/support/knowledge-base/4380-can-i-use-linux-os-with-clickshare-base-units)
+  many of their base unit models can be used with Google Chrome and the Google Cast extension.
```
Comment (SuperSandro2000) [inline nixos/doc/manual/release-notes/rl-2211.section.md:55]:
```suggestion
- The Barco ClickShare driver/client package `pkgs.clickshare-csc1` and the option `programs.clickshare-csc1.enable` have been removed,
  as it requires `qt4`, which reached its end-of-life 2015 and will no longer be supported by nixpkgs.
  [According to Barco](https://www.barco.com/de/support/knowledge-base/4380-can-i-use-linux-os-with-clickshare-base-units) many of their base unit models can be used with Google Chrome and the Google Cast extension.
```

File: pkgs/top-level/aliases.nix [hunk 4]
```diff
@@ -175,6 +175,7 @@ mapAliases ({
   cipherscan = throw "cipherscan was removed from nixpkgs, as it was unmaintained"; # added 2021-12-11
   citra = citra-nightly; # added 2022-05-17
   ckb = throw "'ckb' has been renamed to/replaced by 'ckb-next'"; # Converted to throw 2022-02-22
+  clickshare-csc1 = throw "'clickshare-csc1' has been removed as its dependency qt4 can no longer be maintained"; # Added 2022-06-16
```
Comment (SuperSandro2000) [inline pkgs/top-level/aliases.nix:178]:
```suggestion
  clickshare-csc1 = throw "'clickshare-csc1' has been removed as it requires qt4 which is being removed"; # Added 2022-06-16
```

**Review by SuperSandro2000 — COMMENTED** [review]
Please add aliases.

**Comment by Yarny0** [comment]
Thanks @SuperSandro2000 , I just pushed a new commit which applies all of your suggestions.

> Please add aliases.

I'm not sure what you mean.  Does this refer to the `clickshare-csc1` alias in `aliases.nix` which throws an exception?  Should I add another alias?

**Pull request merged**